index.bs

```diff
@@ -2900,9 +2902,17 @@ of running the following steps:
 
 <h3 id="obtain-aggregatable-report-aggregation-service-payloads">Obtaining an aggregatable report's aggregation service payloads</h3>
 
-To <dfn>obtain the public key for encryption</dfn> given an [=aggregation coordinator=]
-<var ignore=''>aggregationCoordinator</var>, asynchronously return a user-agent-determined public key
-or an error in the event that the user agent failed to obtain the public key.
+To <dfn>obtain the public key for encryption</dfn> given an [=aggregation coordinator=] |aggregationCoordinator|:
+
+1. [=Assert=]: |aggregationCoordinator| is not an [=opaque origin=].
+1. Let |url| be a new [=URL record=].
+1. Set |url|'s [=url/scheme=] to |aggregationCoordinator|'s [=origin/scheme=].
+1. Set |url|'s [=url/host=] to |aggregationCoordinator|'s [=origin/host=].
+1. Set |url|'s [=url/port=] to |aggregationCoordinator|'s [=origin/port=].
+1. Let |path| be «"`.well-known`", "`aggregation-service`", "`public-keys`"».
+1. Set |url|'s [=url/path=] to |path|.
+1. Asynchronously return a user-agent-determined public key from |url| or an error in the event that the user
+    agent failed to obtain the public key from |url|.
```

Comment:
This is vague. Is it really user-agent-determined, or should it reference `[=fetch=]` in some way?

And what does it mean for this to be asynchronous? I think it would be better to let the caller of this algorithm decide to invoke it asynchronously, if needed, rather than inherently require the algorithm to be asynchronous.

Reply:
Hmmm that's what we had before, we can think about how to rephrase it in a follow up?

Reply:
In that case, please add issues for implementing this in terms of fetch and for reorganizing the asynchronous part.

Reply:
Actually rephrased this step a bit. I think the idea is that the public key may be cached in the storage, so it doesn't always have to be fetch.

Reply:
If it can be cached in storage, then we don't even need to construct `url` at all in some cases. I think we will ultimately want to specify fetching the key as one algorithm, and whether to fetch it at all (e.g. how it can be cached) in a different one.

Reply:
Yeah, that makes sense. Added an issue to specify in terms of fetch. Thanks!